pkg/controller/vsphere/reconciler.go

```diff
@@ -389,6 +390,13 @@ func clone(s *machineScope) error {
 	var deviceSpecs []types.BaseVirtualDeviceConfigSpec
 	deviceSpecs = append(deviceSpecs, networkDevices...)
 
+	extraConfig := IgnitionConfig(userData)
+
+	extraConfig = append(extraConfig, &types.OptionValue{
+		Key:   GuestInfoHostname,
+		Value: s.machine.GetName(),
+	})
+
```

Comment:
personally i think this reads misleading and it would be better to write it like,
```suggestion
	extraConfig := []types.BaseOptionValue{}
	extranConfig = append(extraConfig, IgnitionConfig(userData)...)
	extraConfig = append(extraConfig, &types.OptionValue{
		Key:   GuestInfoHostname,
		Value: s.machine.GetName(),
	})

```

this makes it clear that we are collecting the list of options, instead of previously when using the `IgnitionConfig` function it seems like extraConfig is related to ignition configs...

Reply:
Thanks @abhinavdahiya updated